Add openocd dependency to getting F4PGA doc

KennethWilke opened the pull request:
I was hitting some errors with the `make download` examples due to my
lack of openocd, so I think others may find this helpful as well.

umarcor commented:
Since openocd is specific to some boards, I don't think it's desirable to recommend installing it by default. Other users might need other programs for different boards, or even for the same (e.g. openFPGALoader).

Instead, I would suggest to add a "Programming" subsection in Getting F4PGA -> Toolchain installation -> Programming. There, I would explain which tools can be used for programming (which are not installed with the environment).

Furthermore, there is https://f4pga.readthedocs.io/projects/arch-defs/en/latest/getting-started.html#openfpgaloader, which I would like to move into f4pga.readthedocs.io, then reference it from both arch-defs and from here, and extend it to include openocd and others,.

KennethWilke commented:
That makes sense and sounds like a good idea to me. I'm not familiar enough with the ecosystem yet, so not aware at all of which components are necessary and which are specific to boards or vendors or entirely optional. I won't be able to improve this change to offer the better clarity you suggest, but I'd find those details very helpful

kgugala commented:
@KennethWilke can you rebase this PR? We just landed a fix for docs build

KennethWilke committed:
Add openocd dependency to getting F4PGA doc

I was hitting some errors with the `make download` examples due to my
lack of openocd, so I think others may find this helpful as well.
